core/src/main/java/io/parsingdata/metal/data/DataExpressionSource.java

```diff
@@ -63,7 +63,7 @@ public DataExpressionSource(final ValueExpression dataExpression, final int inde
 
     @Override
     public String toString() {
-        return dataExpression.toString() + "[" + index + "](" + graph + "," + encoding + ")";
+        return getClass().getSimpleName() + "(" + dataExpression.toString() + "[" + index + "](" + graph + "," + encoding + "))";
```

Comment:
I agree, also because of performance concerns around this plain string concatenation. But I do propose we do that some other time, as it's not an important change (doesn't affect runtime performance). And about legibility... I don't know.